src/generic/AbsMSeries.jl

```diff
@@ -31,9 +31,19 @@ end
 #
 ###############################################################################
 
+symbols(R::AbsMSeriesRing) = R.sym
+
+parent(a::AbsMSeries) = a.parent
+
+base_ring_type(::Type{<:AbsMSeriesRing{T}}) where T <: RingElement = parent_type(T)
+
+function base_ring(R::AbsMSeriesRing{T}) where T <: RingElement
+    return base_ring(R.poly_ring)::parent_type(T)
```

Comment:
Better call `underlying_poly_ring` instead of direct field access to benefit from the type annotation there. This also applies to all other accesses to this field in this file